Source/Scene/Model.js

```diff
@@ -3774,42 +3855,29 @@ define([
         }
     }
 
-    function modifyShaderForClippingPlanes(shader) {
+    function modifyShaderForClippingPlanes(shader, clippingPlaneCollection) {
         shader = ShaderSource.replaceMain(shader, 'gltf_clip_main');
+        shader += Model._getClippingFunction(clippingPlaneCollection) + '\n';
         shader +=
-            'uniform int gltf_clippingPlanesLength; \n' +
-            'uniform bool gltf_clippingPlanesUnionRegions; \n' +
-            'uniform vec4 gltf_clippingPlanes[czm_maxClippingPlanes]; \n' +
+            'uniform sampler2D gltf_clippingPlanes; \n' +
             'uniform vec4 gltf_clippingPlanesEdgeStyle; \n' +
+            'uniform mat4 gltf_clippingPlanesMatrix; \n' +
             'void main() \n' +
             '{ \n' +
             '    gltf_clip_main(); \n' +
-            '    if (gltf_clippingPlanesLength > 0) \n' +
+            '    float clipDistance = clip(gl_FragCoord, gltf_clippingPlanes, gltf_clippingPlanesMatrix);' +
```

Comment:
Is this specific to glTF or is there a **reasonable** way to encapsulate this so the same GLSL code/function is used for the globe shadre as well?

Reply:
We could re-use this in `PointCloud3DTileContent`, but the logic is baked into `GlobeFS.glsl` and enabled/disabled using injected defines.

I'll go ahead and move it out, it'll be handy in case at some point we want to make all things clippable.